File: Lab/SEM4/MPP/lab6/moto_c/repository/RacerRepository.cs
using moto_c.entity;

namespace moto_c.repository;

public interface RacerRepository : Repository<long, Racer>
{

    List<Racer> findBy(Team team);

    List<Racer> findBy(RaceEvent engine);
    
    void add(Racer racer);
    
    void modify(Racer racer);
}

File: Lab/SEM4/MPP/lab13/moto_java/common/src/main/java/org/zdroba/repository/RacerRepository.java
package org.zdroba.repository;

import org.zdroba.entity.RaceEvent;
import org.zdroba.entity.Racer;
import org.zdroba.entity.Team;

import java.util.List;

public interface RacerRepository extends Repository<Long, Racer> {
    Racer find(String cnp);

    List<Racer> getBy(Team team);

    List<Racer> getBy(RaceEvent engine);

    void add(Racer racer);

    void modify(Racer racer);
}
